src/rgw/rgw_lc.cc

```diff
@@ -43,6 +55,49 @@ void RGWLifecycleConfiguration::_add_rule(LCRule *rule)
   prefix_map[rule->get_prefix()] = rule->get_expiration().get_days();
 }
 
+bool RGWLifecycleConfiguration::check_and_add_rule(LCRule *rule)
+{
+  if (!rule->validate())
+    return false;
+  string id;
+  rule->get_id(id);
+  if (rule_map.find(id) != rule_map.end())  //id shouldn't be the same 
+    return false;
+  rule_map.insert(pair<string, LCRule>(id, *rule));
+
+  auto ret = prefix_map.insert(pair<string, int>(rule->get_prefix(), rule->get_expiration().get_days()));
+  //Now prefix shouldn't be the same. When we add noncurrent expiration or other action, prefix may be same.
+  if (!ret.second)
+    return false;
+  return true;
+}
+
+//Rules are conflicted: if one rule's prefix starts with other rule's prefix, and these two rules
+//define same action(now only support expiration days). 
+bool RGWLifecycleConfiguration::validate() 
```

Comment:
Okay, I see what you're doing here.  I think it should work.  I don't think you need the length check, necessarily, since a string that is a strict initial substring of another will always sort first in C++, but this should still work.